### ISSUE TITLE
The result of monotime-test is unstable

### ISSUE THREAD

**Issue by nshigit** [yugabyte/yugabyte-db#5835]
OS: Ubuntu 18.04.5 LTS x86_64
gcc (Ubuntu 8.4.0-1ubuntu1~18.04) 8.4.0
The result of monotime-test is unstable

YugaBytes: tag v2.3.1.0
```
Build Cmd: ./yb_build.sh --gcc --no-download-thirdparty debug
Test Cmd: /home/nshi/workspace/yugabyte-db/build/debug-gcc-dynamic-ninja/tests-util/monotime-test --gtest_output=xml:/home/nshi/workspace/yugabyte-db/build/debug-gcc-dynamic-ninja/yb-test-logs/tests-util__monotime-test/TestMonoTime_ToCoarse.xml --gtest_filter=TestMonoTime.ToCoarse 
```
TestResut:
```
Note: Google Test filter = TestMonoTime.ToCoarse
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestMonoTime
[ RUN      ] TestMonoTime.ToCoarse
../../src/yb/util/monotime-test.cc:230: Failure
Expected: (converted) <= (after + kPrecision), actual: 8-byte object <B3-9C 1E-07 CA-05 00-00> vs 8-byte object <29-00 0F-07 CA-05 00-00>
Test failure stack trace:
../../src/yb/util/monotime-test.cc:230:                                                                 @     0x55e09ce8d608  yb::TestMonoTime_ToCoarse_Test::TestBody()
[  FAILED  ] TestMonoTime.ToCoarse (187 ms)
[----------] 1 test from TestMonoTime (187 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (187 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestMonoTime.ToCoarse

 1 FAILED TEST

Monotime test could be passed if you run the test several times.

Note: Google Test filter = TestMonoTime.ToCoarse
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestMonoTime
[ RUN      ] TestMonoTime.ToCoarse
[       OK ] TestMonoTime.ToCoarse (0 ms)
[----------] 1 test from TestMonoTime (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (0 ms total)
[  PASSED  ] 1 test.

The source code of failed test case 
------------------------------------------------------------------------
TEST(TestMonoTime, ToCoarse) {
  for (int i = 0; i != 1000; ++i) {
    auto before = CoarseMonoClock::Now();
    auto converted = ToCoarse(MonoTime::Now());
    auto after = CoarseMonoClock::Now();

    // Coarse mono clock has 1ms precision, so we add it to bounds.                                                                                                                                                                            
    const auto kPrecision = 2ms;
    ASSERT_GE(converted, before);
    ASSERT_LE(converted, after + kPrecision);
  }
}
------------------------------------------------------------------------
```
Does the diffirenct compiler or platform could make the implementationof Now() function unstable?

**Comment by nshigit** [comment]
https://www.reddit.com/r/cpp/comments/fmosnb/how_can_anyone_use_stdchronosteady_clock_portably/
The precision of std::chrono::steady_clock is ~3ms, the kPrecision is 2ms in TEST(TestMonoTime, ToCoarse). 
Is Coarse(MonoTime::Now()) function meaningful?

**Comment by es1024** [comment]
Should be fixed by ded5894.
